### ISSUE TITLE
error[E0658]: use of unstable library feature 'stdsimd'

### ISSUE THREAD

**Issue by JameWade** [succinctlabs/sp1#500]
when i run the fibonacci example,i got this err.
![image](https://github.com/succinctlabs/sp1/assets/47301518/03945d1f-4562-4c9c-aeda-9c9a4e65cf8d)

**Comment by DylanCkawalec** [comment]
I would try Updating Rust and Cleaning up Before Reinstallation. 
**That fixed a majority of my problems**. 

**First, update your Rust installation by running:**
`rustup update`
- If there are previously installed packages that might be causing issues, start by uninstalling them:
`cargo uninstall sp1up`
- If the above command doesn’t resolve the issue, navigate to the installation directory and remove the target directory:
r`m -rf target`
Clear the cache:
`rm -rf $HOME/.cargo/bin/cargo-prove`
`rm -rf $HOME/.cargo/registry/cache`


If you initially installed using Homebrew, execute these commands to remove installed files:
`rm ~/.cargo/bin/cargo-prove`
`rm ~/.sp1/bin/cargo-prove`

**Reinstallation from Source**

If cleaning doesn’t help, try reinstalling from source:
```
git clone git@github.com:succinctlabs/sp1.git
cd sp1/cli
cargo install --locked --path .
cd
cargo prove build-toolchain
```
**After building, try running sp1up:**
`sp1up`
**Then install the toolchain:**
`cargo prove install-toolchain`
_Run Through the Procedure Again to build the program and verify it by going between program and script directory's_

[SP1 – Writing Programs Setup Guide](https://succinctlabs.github.io/sp1/writing-programs/setup.html)

**Comment by jtguibas** [comment]
Hi, are you building with AVX512? To run it with AVX512 requires a very specific rust + machine setup.

**Comment by jtguibas** [comment]
This error usually comes from not using a newer rust compiler version that properly supports AVX512 and SIMD.

Please add a `rust-toolchain` file to your `script` folder with the following contents:

```
[toolchain]
channel = "nightly-2024-04-17"
components = ["llvm-tools", "rustc-dev"]
```.

**Comment by RhettCreighton** [comment]
I did rustup update
uninstalled, re-installed cargo-prove using sp1up

when I do:
`
rustc --version
rustc 1.77.0-nightly (7ffc697ce 2024-01-24)

cargo prove new foo

cd foo/script
cat rust-toolchain
[toolchain]
channel = "nightly-2024-01-25"
components = ["llvm-tools", "rustc-dev"]
`

I change to 
nightly-2024-04-17

cargo prove build works in foo/program
in foo/script
`
cargo run --release

error[E0432]: unresolved imports `sp1_sdk::SP1Prover`, `sp1_sdk::SP1Verifier`
 --> src/main.rs:3:15
  |
3 | use sp1_sdk::{SP1Prover, SP1Stdin, SP1Verifier};
  |               ^^^^^^^^^            ^^^^^^^^^^^ no `SP1Verifier` in the root
  |               |
  |               no `SP1Prover` in the root

For more information about this error, try `rustc --explain E0432`.
The following warnings were emitted during compilation:

warning: foo-script@0.1.0: foo-program built at 2024-04-18 19:36:20


`

**Comment by jtguibas** [comment]
Fixed in #542. You will need to run `sp1up` again to get the latest version of the CLI.
